Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs

```diff
@@ -41,59 +41,88 @@ public class TapToPlace : MonoBehaviour, IInputClickHandler
         [Tooltip("Setting this to true will allow this behavior to control the DrawMesh property on the spatial mapping.")]
         public bool AllowMeshVisualizationControl = true;
 
+        private Interpolator interpolator;
+
         /// <summary>
         /// The default ignore raycast layer built into unity.
         /// </summary>
         private const int IgnoreRaycastLayer = 2;
 
-        private Interpolator interpolator;
-
-        private static Dictionary<GameObject, int> defaultLayersCache = new Dictionary<GameObject, int>();
+        private Dictionary<GameObject, int> layerCache = new Dictionary<GameObject, int>();
 
         protected virtual void Start()
         {
             // Make sure we have all the components in the scene we need.
             if (WorldAnchorManager.Instance == null)
             {
                 Debug.LogError("This script expects that you have a WorldAnchorManager component in your scene.");
-            }
-
-            if (WorldAnchorManager.Instance != null)
+            } else if (!IsBeingPlaced)
             {
                 // If we are not starting out with actively placing the object, give it a World Anchor
-                if (!IsBeingPlaced)
-                {
-                    WorldAnchorManager.Instance.AttachAnchor(gameObject, SavedAnchorFriendlyName);
-                }
+                WorldAnchorManager.Instance.AttachAnchor(gameObject, SavedAnchorFriendlyName);
```

Comment:
We should probably still check to make sure `WorldAnchorManager.Instance != null`